### PR TITLE
Even more v2 things.

### PR DESCRIPTION
**Pull request by drichelson**

### CODE REVIEW

File: src/main/java/com/launchdarkly/client/FeatureFlag.java [hunk 1]
```diff
@@ -53,9 +53,19 @@ static FeatureFlag fromJson(String json) {
...
 
   EvalResult evaluate(LDUser user, FeatureStore featureStore) throws EvaluationException {
+    if (user == null || user.getKeyAsString().isEmpty()) {
+      logger.warn("Null user or null/empty user key when evaluating flag: " + key + "; returning null");
+      return null;
+    }
     List<FeatureRequestEvent> prereqEvents = new ArrayList<>();
-    JsonElement value = evaluate(user, featureStore, prereqEvents);
-    return new EvalResult(value, prereqEvents);
+    if (isOn()) {
```
Comment (jkodumal) [inline src/main/java/com/launchdarkly/client/FeatureFlag.java]: What's this change about?

File: src/main/java/com/launchdarkly/client/LDClient.java [hunk 2]
```diff
@@ -174,33 +184,42 @@ private void sendFlagRequestEvent(String featureKey, LDUser user, JsonElement va
...
    * This method will not send analytics events back to LaunchDarkly.
    * <p>
    * The most common use case for this method is to bootstrap a set of client-side feature flags from a back-end service.
    *
    * @param user the end user requesting the feature flags
-   * @return a map from feature flag keys to JsonElement values for the specified user
+   * @return a map from feature flag keys to {@code JsonElement} for the specified user
    */
-  public Map<String, Boolean> allFlags(LDUser user) {
+  public Map<String, JsonElement> allFlags(LDUser user) {
```
Comment (jkodumal) [inline src/main/java/com/launchdarkly/client/LDClient.java]: One step we need to take with the new SDKs is provide a change log / migration guide. This is one of the changes that will be API breaking for customers. Perhaps now is a good time to start creating that doc? 

It should be part of the release notes, but I suggest tabulating them it on quip for now.

**Comment by jkodumal** [comment]
LGTM, but would like to know what that one change was about...

File: src/main/java/com/launchdarkly/client/FeatureFlag.java (hunk 1 above)
Comment (drichelson) [inline src/main/java/com/launchdarkly/client/FeatureFlag.java]: I moved all flag eval logic into the FeatureFlag class. This enables us to call flag.evaluate() in the client's allFLags() method without additional logic around on/off behavior. Also, from an OO perspective it makes more sense: the object contains the data + the behavior.

File: src/main/java/com/launchdarkly/client/LDClient.java [hunk 3]
```diff
@@ -315,22 +334,15 @@ private JsonElement evaluate(String featureKey, LDUser user, JsonElement default
         sendFlagRequestEvent(featureKey, user, defaultValue, defaultValue, null);
         return defaultValue;
       }
-      if (featureFlag.isOn()) {
```
Comment (drichelson) [inline src/main/java/com/launchdarkly/client/LDClient.java]: @jkodumal this logic was moved into the FeatureFlag.evaluate method.

File: src/main/java/com/launchdarkly/client/LDClient.java (hunk 2 above)
Comment (drichelson) [inline src/main/java/com/launchdarkly/client/LDClient.java]: I think the changes should be in the Github release- I'll start a draft for 2.0.0 now.

Comment (drichelson) [inline src/main/java/com/launchdarkly/client/LDClient.java]: https://github.com/launchdarkly/java-client/releases/tag/untagged-e8b0c79a20f1b9ce6469

**Comment by jkodumal** [comment]
👍

**Pull request merged**